dev-java/cglib/cglib-3.1-r1.ebuild

```diff
@@ -0,0 +1,66 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+JAVA_PKG_IUSE="doc examples source test"
+
+inherit java-pkg-2 java-ant-2
+
+DESCRIPTION="cglib is a powerful, high performance and quality Code Generation Library"
+HOMEPAGE="https://github.com/cglib/cglib"
+SRC_URI="mirror://sourceforge/${PN}/${PN}-src-${PV}.jar"
+
+LICENSE="Apache-2.0"
+SLOT="3"
+KEYWORDS="~amd64 ~arm64 ~ppc64 ~x86"
+
+RESTRICT="!test? ( test )"
+
+COMMON_DEP="dev-java/ant-core:0
+	dev-java/asm:4"
+RDEPEND=">=virtual/jre-1.8:*
+	${COMMON_DEP}"
+DEPEND=">=virtual/jdk-1.8:*
+	test? ( dev-java/junit:4 )
+	${COMMON_DEP}"
+
+S="${WORKDIR}"
+
+JAVA_ANT_REWRITE_CLASSPATH="yes"
+EANT_GENTOO_CLASSPATH="asm-4 ant-core"
+
+src_prepare() {
+	default
+
+	# Get rid of some parts in XML.
+	eapply "${FILESDIR}"/${P}-build.xml.patch
+
+	java-pkg_clean
+	cd lib || die 'cannot cd'
+	java-pkg_jarfrom ant-core
+	java-pkg_jarfrom asm-4
+	if use test ; then
+		java-pkg_jarfrom junit-4
```

Comment:
i get this warning when compiling the package:
```
 * java-pkg_ensure-dep: The ebuild is attempting to use junit-4, which is not declared with a SLOT in [RP]DEPEND and --build-only wasn't given.
```

Reply:
Yep, got it.